spec.md

```diff
@@ -2023,18 +2023,47 @@ cover(clk, pred, en, "X equals Y when Z is valid") : optional_name
 
 # Probes
 
-Probe references are created with `probe`{.firrtl} expressions, routed through the design using the `define`{.firrtl} statement, read using the `read`{.firrtl} expression (see [@sec:reading-probe-references]), and forced and released with `force`{.firrtl} and `release`{.firrtl} statements.
+Probes are a feature which facilitate graybox testing of FIRRTL modules.
 
-These statements are detailed below.
+Probes allow modules to expose circuit components contained inside of them.
+Given a probe, you can use it to read the value of the circuit component it references.
+For read-writeable probes, you can also force it, causing it to take on a specified value.
 
-## Define
+Since they are intended for verification, ports with a probe type do not necessarily result in physical hardware.
 
-Define statements are used to route references through the design, and may be used wherever is most convenient in terms of available identifiers -- their location is not significant other than scoping, and do not have last-connect semantics.
-Every sink-flow probe must be the target of exactly one of these statements.
+## Types
 
-The define statement takes a sink-flow static reference target and sets it to the specified reference, which must either be a compatible probe expression or static reference source.
+There are two probe types, `Probe<T>`{.firrtl} is a read-only variant and `RWProbe<T>`{.firrtl} is a read-write variant.
+(See [@sec:probe-types]).
 
-Example:
+Examples:
+
+``` firrtl
+Probe<UInt<8>>
+RWProbe<UInt<8>>
+```
+
+A `RWProbe<T>`{.firrtl} will be implicitly cast to a `Probe<T>`{.firrtl} whenever an expression of type `RWProbe<T>`{.firrtl} is assigned to a port with type `Probe<T>`{.firrtl}.
+
+While `RWProbe<T>`{.firrtl} is strictly more powerful, `Probe<T>`{.firrtl} allows for more aggressive optimization.
+You should consider using `Probe`{.firrtl} whenever you know you don't need to ability to force (see [@sec:forcing]).
+
+## The `probe`{.firrtl} and `rwprobe`{.firrtl} Expressions
+
+Given a reference to a circuit component, you can capture it in a probe with `probe`{.firrtl} for a read-only probe or with `rwprobe`{.firrtl} for a read-writeable probe.
+
+## The `define`{.firrtl} Statement
+
+Since neither `Probe<T>`{.firrtl} nor `RWProbe<T>`{.firrtl} is a connectable type ([@sec:connectable-types]), circuit components with these types cannot be connected to.
+Instead, we use the `define`{.firrtl} statement to route probes through a FIRRTL circuit.
+
+The `define`{.firrtl} statements works similarly to `connect`{.firrtl}, but it has special interactions with conditional statements ([@sec:conditionals]).
+`define`{.firrtl} statements may appear within conditional blocks, and when they do, they may connect circuit components defined in a local scope.
```

Comment:
```suggestion
`define`{.firrtl} statements may appear within conditional blocks, and when they do, they may refer to circuit components defined in a local scope.
```
(mostly, probably should avoid saying `define` "connects" anything? And I think idea here is answering the question "why would you use define from within a conditional?" and the answer is "so the things you're referring to are in-scope" (right?))